samples/basic/blinky/src/main.c

```diff
@@ -33,7 +35,7 @@ void main(void)
 	bool led_is_on = true;
 	int ret;
 
-	dev = device_get_binding(LED0);
+	dev = device_if_ready(DEVICE_DT_GET(CTLR_NODE));
```

Comment:
Doesn't this break until we convert all the GPIO drivers over to DEVICE_DT_DEFINE?

It seems like we should merge this PR without this change, update all the drivers in another PR or set of PRs, then convert some things in samples/basic over to this new API after the drivers are updated.

Otherwise I think this looks good and is a massive improvement.

When all the drivers are converted, we can also update https://docs.zephyrproject.org/latest/guides/dts/howtos.html#get-a-struct-device-from-a-devicetree-node to reflect the new approach.

Reply:
> Doesn't this break until we convert all the GPIO drivers over to DEVICE_DT_DEFINE?

Yes, though it's surprising only `frdm_k64f` fails CI.

I'm not sure whether to mark this DNM, hope somebody does a Changes Requested pending removal of the commit, or remove the commit now and deal with concerns about why there aren't examples when they arise.

Reply:
I'd say drop the change to the sample for now and we can deal with it once we've got the full class of drivers converted over.  I'm guessing we'll end up introducing a helper that does both the DEVICE_DT_GET and ready check at the same time.

Reply:
> I'm not sure whether to mark this DNM, hope somebody does a Changes Requested pending removal of the commit, or remove the commit now and deal with concerns about why there aren't examples when they arise.

I thought we discussed ginning up some fake devices in the test suite for the new instantiation macro during dev review?

Reply:
For  #29644 and the dependency API, yes.  I thought this part was sufficiently clear and non-controversial to avoid that make-work, and that your bme280 example could be the proof-of-validation if one was necessary.

I've removed the blinky example.  For the record: here's what API use looks like:

```
diff --git a/samples/basic/blinky/src/main.c b/samples/basic/blinky/src/main.c
index 33ed9c3f58..e2fbfd3dc5 100644
--- a/samples/basic/blinky/src/main.c
+++ b/samples/basic/blinky/src/main.c
@@ -16,8 +16,6 @@
 #define LED0_NODE DT_ALIAS(led0)
 
 #if DT_NODE_HAS_STATUS(LED0_NODE, okay)
-#define CTLR_NODE DT_GPIO_CTLR(LED0_NODE, gpios)
-DEVICE_DT_DECLARE(CTLR_NODE);
 #define LED0   DT_GPIO_LABEL(LED0_NODE, gpios)
 #define PIN    DT_GPIO_PIN(LED0_NODE, gpios)
 #define FLAGS  DT_GPIO_FLAGS(LED0_NODE, gpios)
@@ -35,7 +33,7 @@ void main(void)
        bool led_is_on = true;
        int ret;
 
-       dev = device_if_ready(DEVICE_DT_GET(CTLR_NODE));
+       dev = device_get_binding(LED0);
        if (dev == NULL) {
                return;
        }
```

Reply:
> I've removed the blinky example.

Sounds reasonable